partiql-catalog/src/scalar_fn.rs

```diff
@@ -0,0 +1,82 @@
+use crate::call_defs::{CallSpecArg, ScalarFnCallDef, ScalarFnCallSpec};
+use crate::context::SessionContext;
+
+use crate::extension::ExtensionResultError;
+use dyn_clone::DynClone;
+use partiql_common::FN_VAR_ARG_MAX;
+use partiql_value::Value;
+use std::borrow::Cow;
+use std::fmt::{Debug, Formatter};
+
+pub type ScalarFnExprResultValue<'a> = Cow<'a, Value>;
+pub type ScalarFnExprResult<'a> = Result<ScalarFnExprResultValue<'a>, ExtensionResultError>;
+
```

Comment:
It can't. 
`partiql-eval` imports `partiql-catalog`. Moving them to `eval` would necessitate `catalog` importing `eval` which would create a cycle.

For this reason, I tried to keep anything other than `Value` out of the signature.

Reply:
I see, how about `partiql-commons`?